Ginger/GingerCoreNET/Repository/SharedRepositoryOperations.cs

```diff
@@ -122,9 +122,30 @@ public Boolean UploadItemToRepository(Context context, UploadItemSelection itemT
                 {
                     itemToUpload.UsageItem.ParentGuid = Guid.Empty;
                 }
-                if (itemToUpload.ReplaceType == UploadItemSelection.eActivityInstanceType.LinkInstance && !itemToUpload.UsageItem.IsLinkedItem)
+                if (itemToUpload.ReplaceType == UploadItemSelection.eActivityInstanceType.LinkInstance && !itemToUpload.UsageItem.IsLinkedItem && itemCopy is Activity)
                 {
                     context.BusinessFlow.MarkActivityAsLink(itemToUpload.ItemGUID, itemCopy.Guid);
+                    Activity referenceActivity = (Activity)itemCopy;
+                    Activity activity = context.BusinessFlow.Activities.FirstOrDefault(a => a.Guid == itemToUpload.ItemGUID);
```

Comment:
Setting parent GUID should be same for Link and Regular instance.

Reply:
added the logic above for both Linked and Regular case